tap_zoom/sync.py

```diff
@@ -9,28 +10,143 @@
 
 LOGGER = singer.get_logger()
 
-def get_bookmark(state, stream_name, default):
-    return state.get('bookmarks', {}).get(stream_name, default)
-
-def write_bookmark(state, stream_name, value):
-    if 'bookmarks' not in state:
-        state['bookmarks'] = {}
-    state['bookmarks'][stream_name] = value
-    singer.write_state(state)
 
 def write_schema(stream):
     schema = stream.schema.to_dict()
     singer.write_schema(stream.tap_stream_id, schema, stream.key_properties)
 
 
+def update_key_bag_for_child(key_bag, parent_endpoint, record):
+    # Constructs the properties needed to build the nested
+    # paths used by the Zoom APIs.
+    # Ex. the list of recordings is fetched at
+    #   /users/{userId}/recordings
+    # We get the list of user records and pass the following
+    # key bag to the recordings endpoint:
+    #   {"userId": <user_id>}
+    updated_key_bag = dict(key_bag)
+
+    if parent_endpoint and 'provides' in parent_endpoint:
+        for dest_key, obj_key in parent_endpoint['provides'].items():
+            updated_key_bag[dest_key] = record[obj_key]
+
+    return updated_key_bag
+
+
+def sync_recordings(client,
+                    catalog,
+                    state,
+                    required_streams,
+                    selected_streams,
+                    stream_name,
+                    endpoint,
+                    key_bag,
+                    parent_endpoint=None,
+                    records=[]):
+    utc_format = '%Y-%m-%d'
+    start_date = singer.get_bookmark(state,
+                                     stream_name,
+                                     'endDate',
+                                     client.start_date)
+    current_date = datetime.utcnow().date()
+    if start_date:
+        start_datetime = singer.utils.strptime_to_utc(start_date)
+    else:
+        # If no start_date or bookmark available, default to
+        # ingesting yesterday's data.
+        yesterday = current_date - timedelta(days=1)
+        start_datetime = singer.utils.strptime_to_utc(yesterday.strftime(utc_format))
```

Comment:
same here, yesterday is datelike after calling yesterday.strftime(utc_format).

Reply:
related to comment about unused config start_date, it should be used here right?